module/fallbackstrategy.go

```diff
@@ -0,0 +1,11 @@
+package module
+
+// FallbackStrategy defines an interface to handle usage of multiple
+// clients connected to different hosts.
+type FallbackStrategy interface {
+	// ClientIndex return index of client to use
+	ClientIndex() int
+
+	// Failure indicate failure to use client at active index
+	Failure()
```

Comment:
We should include the client index being used here, otherwise we risk a race condition. The user of `FallbackStrategy` wants to tell `FallbackStrategy`, "I tried to talk to client X but it didn't work". If we don't pass in "X", which client we're referring to, we are requiring implicit synchronization between `FallbackStrategy`'s internal state and its user's state.